static/app/components/integrationExternalMappingForm.tsx

```diff
@@ -4,127 +4,142 @@ import capitalize from 'lodash/capitalize';
-        onResults: result => {
-          // For organizations with >100 teams, we want to make sure their
-          // saved mapping gets populated in the results if it wouldn't have
-          // been in the initial 100 API results, which is why we add it here
-          if (mapping && !result.find(({id}) => id === mapping.teamId)) {
-            result = [{id: mapping.teamId, name: mapping.sentryName}, ...result];
-          }
-          // The team needs `this.props.onResults` so that we have team slug
-          // when a user submits a team mapping, the endpoint needs the slug
-          // as a path param: /teams/${organization.slug}/${team.slug}/external-teams/
-          this.props.onResults?.(result);
-          return optionMapper(sentryNamesMapper(result));
-        },
+        label: isInline ? undefined : tct('External [type]', {type: capitalize(type)}),
+        placeholder: t(`${type === 'user' ? '@username' : '@org/teamname'}`),
       });
     }
     return fields;
   }
 
-  render() {
-    const {onSubmitSuccess, onCancel, mapping, baseEndpoint, onSubmit} = this.props;
+  get extraFormFieldProps() {
+    const {isInline} = this.props;
+    return isInline
+      ? {
+          // We need to submit the entire model since it could be a new one or an update
+          getData: () => this.model.getData(),
+          // We need to update the model onBlur for inline forms since the model's 'onPreSubmit' hook
+          // does NOT run when using `saveOnBlur`.
+          onBlur: () => this.updateModel(),
+        }
+      : {flexibleControlStateSize: true};
+  }
 
-    // endpoint changes if we are making a new row or updating an existing one
-    const endpoint = !baseEndpoint
-      ? undefined
-      : mapping
-      ? `${baseEndpoint}${mapping.id}/`
-      : baseEndpoint;
-    const apiMethod = !baseEndpoint ? undefined : mapping ? 'PUT' : 'POST';
+  // This function is necessary since the endpoint we submit to changes depending on the value selected
+  updateModel() {
+    const mapping = this.model.getData() as ExternalActorMapping;
+    const {getBaseFormEndpoint} = this.props;
+    if (mapping) {
+      const endpointDetails = getExternalActorEndpointDetails(
+        getBaseFormEndpoint(mapping),
+        mapping
+      );
+      this.model.setFormOptions({...this.model.options, ...endpointDetails});
+    }
+  }
 
+  render() {
+    const {isInline, onCancel, onSubmitError, onSubmitSuccess} = this.props;
     return (
       <FormWrapper>
         <Form
           requireChanges
-          onSubmitSuccess={onSubmitSuccess}
+          model={this.model}
           initialData={this.initialData}
-          apiEndpoint={endpoint}
-          apiMethod={apiMethod}
           onCancel={onCancel}
-          onSubmit={onSubmit}
+          onSubmitSuccess={onSubmitSuccess}
+          onSubmitError={onSubmitError}
+          saveOnBlur={isInline}
+          allowUndo={isInline}
+          onPreSubmit={() => this.updateModel()}
```

Comment:
Same as what we specified above in `onBlur`. We have to do it twice because like I said up there, `onPreSubmit` does not fire for `saveOnBlur` forms.